.azure-pipelines/testbed_health_check.py

```diff
@@ -303,22 +310,40 @@ def check_bgp_session_state(self, state="established"):
         logger.info("======================= check_bgp_session_state starts =======================")
 
         for sonichost in self.sonichosts:
+            if (self.is_multi_asic and
+                    self.duts_basic_facts[sonichost.hostname]["ansible_facts"]["dut_basic_facts"]["is_supervisor"]):
+                logger.info("Skip check_bgp_session_state on Supervisor.")
+                continue
 
             hostname = sonichost.hostname
 
             logger.info("----------------------- check_bgp_session_state on [{}] -----------------------".format(
                 hostname))
 
             # Retrieve BGP facts for the Sonic host
-            bgp_facts = sonichost.bgp_facts()['ansible_facts']
+            if self.is_multi_asic:
+                bgp_facts = {}
+                host_asics_list = self.duts_basic_facts[sonichost.hostname][
+                    "ansible_facts"]["dut_basic_facts"]["asic_presence_list"]
+
+                for instance_id in host_asics_list:
+                    bgp_facts[instance_id] = sonichost.bgp_facts(instance_id=instance_id)['ansible_facts']
+            else:
+                bgp_facts = sonichost.bgp_facts()['ansible_facts']
 
             bgp_facts_on_hosts[hostname] = bgp_facts
 
             # Check BGP session state for each neighbor
             neigh_not_ok = []
-            for k, v in list(bgp_facts['bgp_neighbors'].items()):
-                if v['state'] != state:
-                    neigh_not_ok.append(f"{k}, {v['state']}")
+            if self.is_multi_asic:
+                for instance_id in bgp_facts:
+                    for k, v in list(bgp_facts[instance_id]['bgp_neighbors'].items()):
+                        if v['state'] != state:
+                            neigh_not_ok.append(f"{k}, {v['state']}")
+            else:
+                for k, v in list(bgp_facts['bgp_neighbors'].items()):
+                    if v['state'] != state:
+                        neigh_not_ok.append(f"{k}, {v['state']}")
```

Comment:
The code looks like duplicate, can we use bgp_facts = {} for both single asic and multi-asic?